### PR TITLE
Extracted core as package and fixed ts config

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

File: packages/app/src/core/models/Space/Space.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import {Instance, types, flow, cast} from 'mobx-state-tree';
+import {UUIDModel} from '@momentum/core/dist/models/UUID';
+import {RequestModel} from '@momentum/core/dist/models/Request';
+import {ResetModel} from '@momentum/core/dist/models/ResetModel';
```
Comment (dmitry-yudakov) [inline packages/app/src/core/models/Space/Space.ts:4]: something got out of control here :)

File: packages/app/src/core/models/SpaceUser/SpaceUserModel.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import {Instance, types} from 'mobx-state-tree';
+import {UUIDModel} from '@momentum/core/dist/models/UUID';
```
Comment (dmitry-yudakov) [inline packages/app/src/core/models/SpaceUser/SpaceUserModel.ts:2]: and here

File: packages/app/src/core/models/UserProfile/UserProfileModel.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import {types, Instance, flow} from 'mobx-state-tree';
+import {UUIDModel} from '@momentum/core/dist/models/UUID';
```
Comment (dmitry-yudakov) [inline packages/app/src/core/models/UserProfile/UserProfileModel.ts:2]: and here

**Review by dmitry-yudakov — APPROVED** [review]
Impressive work, well done :)

**Review by aminbksi — APPROVED** [review]

**Pull request merged**